fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAnalyzeStmt.java

```diff
@@ -0,0 +1,366 @@
+        return tblIds;
+    }
+
+    public String getStateValue() {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The tbl name must be obtained after the parsing is complete");
+        return stateValue;
+    }
+
+    public ArrayList<OrderByPair> getOrderByPairs() {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The tbl name must be obtained after the parsing is complete");
+        return orderByPairs;
+    }
+
+    public long getLimit() {
+        if (limitElement != null && limitElement.hasLimit()) {
+            return limitElement.getLimit();
+        }
+        return -1L;
+    }
+
+    public long getOffset() {
+        if (limitElement != null && limitElement.hasOffset()) {
+            return limitElement.getOffset();
+        }
+        return -1L;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
+        super.analyze(analyzer);
+
+        if (dbTableName != null) {
+            dbTableName.analyze(analyzer);
+            String dbName = dbTableName.getDb();
+            String tblName = dbTableName.getTbl();
+            checkShowAnalyzePriv(dbName, tblName);
+
+            Database db = analyzer.getEnv()
+                    .getInternalDataSource().getDbOrAnalysisException(dbName);
+            Table table = db.getTableOrAnalysisException(tblName);
+            dbId = db.getId();
+            tblIds.add(table.getId());
+        } else {
+            // analyze the current default db
+            String dbName = analyzer.getDefaultDb();
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+
+            Database db = analyzer.getEnv()
+                    .getInternalDataSource().getDbOrAnalysisException(dbName);
+
+            db.readLock();
+            try {
+                List<Table> tables = db.getTables();
+                for (Table table : tables) {
+                    checkShowAnalyzePriv(dbName, table.getName());
+                }
+
+                dbId = db.getId();
+                for (Table table : tables) {
+                    long tblId = table.getId();
+                    tblIds.add(tblId);
+                }
+            } finally {
+                db.readUnlock();
+            }
+        }
+
+        // analyze where clause if not null
+        if (whereClause != null) {
+            if (whereClause instanceof CompoundPredicate) {
+                CompoundPredicate cp = (CompoundPredicate) whereClause;
+                if (cp.getOp() != CompoundPredicate.Operator.AND) {
+                    throw new AnalysisException("Only allow compound predicate with operator AND");
+                }
+                // check whether left.columnName equals to right.columnName
+                checkPredicateName(cp.getChild(0), cp.getChild(1));
+                analyzeSubPredicate(cp.getChild(0));
```

Comment:
only support one level CompoundPredicate?

Reply:
for statistical job, I think the user mainly cares which state it belongs to, and I've simplified it.